src/utils/makeReactNativeConfig.js

```diff
@@ -192,7 +192,7 @@ const getDefaultConfig = ({
         new HasteResolver({
           directories: providesModuleNodeModules
             ? providesModuleNodeModules.map(_ => moduleResolve(root, _))
-            : [moduleResolve(root, 'react-native')],
+            : [path.join(moduleResolve(root, 'react-native'), 'Libraries')],
```

Comment:
Something more is needed here.  
For users that specify a custom list of providesModuleNodeModules we need to ensure that at least react-native still gets scoped to the Libraries directory.

I wonder if providesModuleNodeModules option should be expanded to be able to include specific directories within the module.

Something like changing the type of providesModuleNodeModules to: ?
providesModuleNodeModules: (string | {module: string; directories: string[]})[]

Reply:
I changed the code locally to: 
```
        new HasteResolver({
          directories: (providesModuleNodeModules || [
            'react-native',
          ]).map(_ => {
            if (typeof _ === 'string') {
              if (_ === 'react-native') {
                return path.join(
                  moduleResolve(root, 'react-native'),
                  'Libraries'
                );
              }
              return moduleResolve(root, _);
            }
            return path.join(moduleResolve(root, _.module), _.directory);
          }),
          hasteOptions: hasteOptions || {},
        })
```
Which I think keeps the existing configs working, and allows additional modules to specify specific directories too.